src/app/modules/forms/ix-forms/components/ix-input/ix-input.component.ts

```diff
@@ -153,9 +157,15 @@ export class IxInputComponent implements ControlValueAccessor, OnInit, OnChanges
   }
 
   registerOnChange(onChange: (value: string | number) => void): void {
-    this.onChange = (val) => {
-      this.lastKnownValue = val;
-      onChange(val);
+    this.onChange = (changedValue) => {
+      if (this.lastKnownValue !== changedValue) {
+        onChange(changedValue);
+      }
+
+      this.controlDirective.control.markAsTouched();
```

Comment:
It shouldn't be up to you to trigger this. The way for controls to interact with angular forms is via `onChange`/`onTouch` functions that are provided. If there was no change, we should avoid calling `onChange`.